File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string r = "";
        int number;
        // ab // rqw
        if(word1.length() > word2.length()){
            number = word1.length();
        }else{
            number = word2.length();
        }
        for(int i = 0; i < number ; i++){

            if(word1.length() > i){
                r += word1[i];
            }
            if(word2.length()> i){
                r += word2[i];
            }

        }
        return r;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        F, S = 0,0
        res = ""

        while F < len(word1) and S < len(word2):
            res += word1[F]
            res += word2[S]
            F+= 1
            S+= 1
        
        if F < len(word1):
            res += word1[F:]
        else:
            res += word2[S:]
        return res




        